src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -1449,6 +1459,17 @@ BlittableJsonReaderObject UpdateDatabaseRecordIfNeeded()
             }
         }
 
+        private static bool IsClientConfigChanged(BlittableJsonReaderObject newDatabaseRecord, BlittableJsonReaderObject oldDatabaseRecord)
+        {
+            const string clientPropName = nameof(DatabaseRecord.Client);
+            if (newDatabaseRecord.TryGet(clientPropName, out BlittableJsonReaderObject newDbClientConfig) == false || newDbClientConfig == null) 
```

Comment:
Not sure that I follow. If new record does not have a configuration then it didn't change (assuming that the old one has) ?

Reply:
The goal is to give the new client configuration the ETag of the database record.
And to do that only if the client configuration changed.
If there is no new client configuration so there is no configuration to set the ETag.
Maybe if I will change the method name (and variable) to `ShouldSetClientConfigEtag` or something similar it will be more clear?

Reply:
Let's go with that. Yes.